### ISSUE TITLE
Annotation editing support in toolbox

### ISSUE THREAD

**Issue by jeznag** [apache/incubator-echarts#10020]
### What problem does this feature solve?
Currently there is no easy way for a user to add annotations to a chart. The data can be initialised with labelled points but for my use case, I need my data analysts to be able to highlight a chart area and directly add annotations in a similar way to http://demo.blacklabel.pl/ (see the buttons in the top right: text, rectangle)

### What does the proposed API look like?
1. The toolbox would be able to be initialised with addAnnotations: true - this would add buttons in the top right corner similar to http://demo.blacklabel.pl/

2. When annotations are added, an event would be triggered with a payload like this:
```
{
   annotationAdded: { annotationType: 'rectangle', annotationText: 'blah', annotationXMin: 123, annotationXMax: 150, annotationYMin: 100, annotationYMax: 200 },
   allAnnotations: [{ annotationType: 'rectangle', annotationText: 'blah', annotationXMin: 123, annotationXMax: 150, annotationYMin: 100, annotationYMax: 200 }]
}
```

**Comment by Ovilia** [comment]
Yes. We are considering enhance labeling feature with annotation. There shall be much work to do, and it is scheduled in ECharts v5, which we haven't begin with yet.

**Comment by hushaoqing** [comment]
Looking forward to this feature !

**Comment by alexapluss** [comment]
Hi Olivia,

Do you have an update on when this will be available? Also in general, when is 5.0.0 to be released

**Comment by msbasanth** [comment]
Looking forward for annotations feature which will allow users to annotate a chart freely with labels and shapes.

**Comment by nouman0320** [comment]
is this feature still in pipeline?

**Comment by yutingli119** [comment]
still waiting for this feature in 2021....

**Comment by Ovilia** [comment]
https://echarts.apache.org/examples/zh/editor.html?c=scatter-label-align-top Is this what you want? @msbasanth @nouman0320 @yutingli119

**Comment by msbasanth** [comment]
@Ovilia The static annotations are good. 

Whether chart user can add annotations dynamically for marking/highlighting directly in the chart using tools provided with which we can add annotations. This will be a really good to have feature.

**Comment by yutingli119** [comment]
> @Ovilia The static annotations are good.
> 
> Whether chart user can add annotations dynamically for marking/highlighting directly in the chart using tools provided with which we can add annotations. This will be a really good to have feature.

Agreed. That's what I was looking forward to have...

**Comment by pissang** [comment]
Currently we don't provide such a toolbox. But it's not so complex to combine the echarts event, setOption update and labelLine features to get something similar.

Here is a example about using these features to interactive with the chart programtically
https://echarts.apache.org/examples/zh/editor.html?c=line-draggable

**Comment by tomasz-zablocki-ziegert** [comment]
It's the only missing feature that makes our team hesitant about adopting echarts for our product. It's already a majority of our customers that made it clear that it's a mandatory to have it.

**Comment by kchjxxgh** [comment]
Is there any 3rd extension to support such interactive behavior, i.e. user add annotation and add line. 
If no, I assume in the future, echarts framework level should make some interactive foundation, so that the 3rd developer can do their own various line based on such foundation/api, just like like highstocks, toolbox can support shape, Fibonacci line,  Pitchfork line, Parallel channel line and so on.

**Comment by MrSquaare** [comment]
Any news about this feature?
Maybe echarts could integrate [fabric.js](https://github.com/fabricjs/fabric.js) to do this

**Comment by robertabenassi** [comment]
I find echarts a very nice plotting library.  This feature to be a built-in feature will be a great boost in the application I manage

**Comment by vikas-6** [comment]
> https://echarts.apache.org/examples/zh/editor.html?c=scatter-label-align-top Is this what you want? @msbasanth @nouman0320 @yutingli119

https://codepen.io/VIKAS-KUMAR-the-vuer/pen/LYXWWXz?editors=0010

**Comment by natasha-f** [comment]
It would be much appreciated if echarts had annotations feature similar to this:
https://www.highcharts.com/docs/advanced-chart-features/annotations-module
